src/templateConditionalComplexityRule.ts

```diff
@@ -0,0 +1,562 @@
+
+        if (treeType === true) {
+          // Create one evalPath where every condition is true
+          this.evalPaths = this.generateEvalPathsUniformResult(expression.conditions, true);
+        } else {
+          // Create a separate falsePath for each Condition where one is false
+          this.evalPaths = this.generateEvalPathsIndividualResult(expression.conditions, false);
+        }
+
+      } else if (expression.operators[0].isOr()) {
+
+        if (treeType === true) {
+          // Create a separate evalPath for each Condition where one is true
+          this.evalPaths = this.generateEvalPathsIndividualResult(expression.conditions, true);
+        } else {
+          // Create one falsePath where every condition is false
+          this.evalPaths = this.generateEvalPathsUniformResult(expression.conditions, false);
+        }
+
+      } else { // XOR operator
+
+        // A XOR expression is true if it has an odd number of true conditions
+        const isTrueCondition = (c) => {
+            return c.result;
+          },
+          hasOddTrues = (ep) => {
+            return ep.filter(isTrueCondition).length % 2 === 1;
+          },
+          allEvalPaths = this.calculateAllEvalPaths(expression.conditions);
+
+        allEvalPaths.forEach((evalPath) => {
+          if (hasOddTrues(evalPath) !== treeType === false) {
+            this.evalPaths.push(evalPath);
+          }
+        });
+
+      }
+    }
+  }
+
+
+  private calculateAllEvalPaths(conditions): Array<any> {
+    // recursively calculate the eval paths of all the subsequent elements
+    let subEvalPaths: Array<any> = (conditions.length === 1) ? [[]] : this.calculateAllEvalPaths(conditions.slice(1)),
+      evalPaths = [],
+      tempEvalPath;
+
+    // For each of the sub paths, create versions where this condition is true and false
+    subEvalPaths.forEach((e) => {
+      (tempEvalPath = Utils.mergeDeep(e)).unshift({condition: conditions[0], result: true});
+      evalPaths.push(tempEvalPath);
+      (tempEvalPath = Utils.mergeDeep(e)).unshift({condition: conditions[0], result: false});
+      evalPaths.push(tempEvalPath);
+    });
+
+    return evalPaths;
+  }
+
+  // Generate eval paths where each condition has the specified result
+  private generateEvalPathsIndividualResult(conditions, result): Array<any> {
+    let evalPaths = [], evalPath;
+    conditions.forEach((unused, i1) => {
+      evalPath = [];
+      conditions.forEach((c, i2) => {
+        evalPath.push({condition: c, result: (i1 === i2) ? result : null});
+      });
+      evalPaths.push(evalPath);
+    });
+    return evalPaths;
+  }
+
+  // Generate an eval path with the specified result for all conditions
+  private generateEvalPathsUniformResult(conditions, result): Array<any> {
+    let evalPaths = [[]];
+    conditions.forEach((c) => {
+      evalPaths[0].push({condition: c, result: result});
+    });
+    return evalPaths;
+  }
+
+
+}
+
+class Expression {
```

Comment:
Curious why you didn't choose to use the expression parser of Angular?

Reply:
I didn't use the parser for three main reasons:
- I didn't find easily the information I needed. I can't say that this parser is very "user friendly".
- Without the expression parser of Angular, we are bound only on the public API.
- Our analysis is purely static

Reply:
If we use custom parser we can hit a problem because of the DSL Angular supports, which has extra syntax on top of JavaScript.

We already depend on the expression parser in the `NgWalker`. I don't think using it will make our analysis dynamic?

What information is missing there?

Reply:
Hi @mgechev,
Ok, I've found my holy graal ! thx.

Reply:
Beautiful! Will take a final look tomorrow and merge.